tools/oprt2/pkg/packagemanager/apt/apt.go

```diff
@@ -0,0 +1,243 @@
+// including the file's path. For example, `some/path/teleport.deb`.
+func WithComponents(components map[string][]*regexp.Regexp) APTOption {
+	return func(apt *APT) {
+		if len(apt.components) == 0 {
+			apt.components = make(map[string][]*regexp.Regexp, len(components))
+		}
+
+		// Deep merge the maps
+		for componentName, fileNameMatchers := range components {
+			if _, ok := apt.components[componentName]; !ok {
+				components[componentName] = fileNameMatchers
+				continue
+			}
+			components[componentName] = append(apt.components[componentName], fileNameMatchers...)
+		}
+
+		// Get unique matchers
+		// This does not preserve order, but it shouldn't need to
+		for componentName, fileNameMatchers := range components {
+			uniqueMatchers := make(map[string]*regexp.Regexp, len(fileNameMatchers))
+			for _, fileNameMatcher := range fileNameMatchers {
+				uniqueMatchers[fileNameMatcher.String()] = fileNameMatcher
+			}
+
+			if len(uniqueMatchers) != len(fileNameMatchers) {
+				keys := slices.Collect(maps.Keys(uniqueMatchers))
+				slices.Sort(keys)
+
+				sortedUniqueMatchers := make([]*regexp.Regexp, 0, len(keys))
+				for _, key := range keys {
+					sortedUniqueMatchers = append(sortedUniqueMatchers, uniqueMatchers[key])
+				}
+
+				components[componentName] = sortedUniqueMatchers
+			}
+		}
+
+		apt.components = components
+	}
+}
+
+// WithDistros configures the package manager to add packages to the specified distros. Key
+// is the distro name (e.g. ubuntu, debian), value is distro version (e.g. noble, plucky)
+func WithDistros(distros map[string][]string) APTOption {
+	return func(apt *APT) {
+		if len(apt.distros) == 0 {
+			apt.distros = make(map[string][]string, len(apt.distros))
+		}
+
+		// Deep merge the maps
+		for distroName, distroVersion := range distros {
+			if _, ok := apt.distros[distroName]; !ok {
+				distros[distroName] = distroVersion
+				continue
+			}
+			distros[distroName] = append(apt.distros[distroName], distroVersion...)
+		}
+
+		// Get unique values only
+		for distroName, distroVersions := range distros {
+			uniqueVersions := make(map[string]struct{}, len(distroVersions))
+			for _, distroVersion := range distroVersions {
+				uniqueVersions[distroVersion] = struct{}{}
+			}
+
+			// Collect the results if anything has changed
+			if len(uniqueVersions) != len(distroVersions) {
+				distros[distroName] = slices.Collect(maps.Keys(uniqueVersions))
+			}
+
+			// Sort to make the process a little more predictable (e.g. log output ordering)
+			slices.Sort(distros[distroName])
+		}
+
+		apt.distros = distros
+	}
+}
+
+// WithLogger configures the package manager with the provided logger.
+func WithLogger(logger *slog.Logger) APTOption {
+	return func(apt *APT) {
+		if logger == nil {
+			logger = logging.DiscardLogger
+		}
+		apt.logger = logger
+	}
+}
+
+// NewAPT creates a new APT package manager instance.
+func NewAPT(fm filemanager.FileManager, opts ...APTOption) *APT {
```

Comment:
Nitpick but as a rule of thumb, I'd keep the NewX method at the top of the file.